### PR TITLE
Unset PYTHONPATH env var

### PR DESCRIPTION
**Pull request by albertvaka**
### What does this PR do?

Unset PYTHONPATH env var before initializing the embedded Python runtime.

### Motivation

On Windows, it's not uncommon to set PYTHONPATH. If set, it would be picked up by the embedded Python, mixing our bundled site-packages with the system ones (and likely breaking things).

### CODE REVIEW

File: pkg/collector/python/init.go [hunk 1]
```diff
@@ -315,6 +315,12 @@ func Initialize(paths ...string) error {
 		C.add_python_path(rtloader, TrackedCString(p))
 	}
 
+	// On Windows, it's not uncommon to have a system-wide PYTHONPATH env var set.
+	// Remove it, so our embedded python doesn't try to load things from the system.
+	if runtime.GOOS == "windows" && !config.Datadog.GetBool("windows_use_pythonpath") {
```
Comment (truthbk) [inline pkg/collector/python/init.go]: I would put this here instead: https://github.com/DataDog/datadog-agent/blob/albertvaka/unset-pythonpath/pkg/collector/python/init_windows.go#L12

**Review by truthbk — COMMENTED** [review]
small suggestion regarding init code

File: pkg/collector/python/init.go (hunk 1 above)
Comment (albertvaka) [reply pkg/collector/python/init.go]: makes sense 👍

**Review by olivielpeau — DISMISSED** [review]
LGTM, thanks for leading this discussion + implementing

Should we document in the config_template as well? (I'd vote "yes" unless there's a reason not to)

**Review by derekwbrown — APPROVED** [review]

**Pull request merged**